gateway/ln-gateway/src/gatewayd/types.rs

```diff
@@ -0,0 +1,42 @@
+//! Map `gatewaylnrpc` protobuf types to rust types
+
+use secp256k1::PublicKey;
+
+impl From<crate::gatewaylnrpc::get_route_hints_response::RouteHintHop>
+    for mint_client::modules::ln::route_hints::RouteHintHop
+{
+    fn from(hop: crate::gatewaylnrpc::get_route_hints_response::RouteHintHop) -> Self {
+        let slice: &[u8; 33] = &hop
+            .src_node_id
+            .try_into()
+            .expect("malformed source node id");
```

Comment:
> Note: This trait must not fail.  [rust doc](https://doc.rust-lang.org/std/convert/trait.From.html)

I think  [TryFrom](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) would be better suited here

Reply:
This is excellent :)